### PR TITLE
Don't let the IDE throw away executionContext/executionStatus messages

### LINKED ISSUE #5260: Throwing an unhandled exception in Polyglot Java results in a null in the UI and no caught error message
This task is automatically imported from the old Task Issue Board and it was originally created by James Dunkerley.
Original issue is [here](https://www.pivotaltracker.com/story/show/184073253).
<hr />

Throwing a NPE within Java caused an error when command line:
```
Execution finished with an error: null
        at <java> org.enso.table.data.index.MultiValueIndex.makeCrossTabTable(MultiValueIndex.java:185)
        at <enso> case_branch(C:\Repos\Enso\ide\built-distribution\enso-engine-0.0.0-dev-windows-amd64\enso-0.0.0-dev\lib\Standard\Table\0.0.0-dev\src\Data\Table.enso:1299:21-122)
```

Within the UI get no error message and the viz shows `null`. Error from Language Server shown below.

## Comments:
 (James Dunkerley - Dec 20, 2022)
<hr />
```
from Standard.Base import Panic
polyglot java import java.lang.NullPointerException

main =
    x = NullPointerException.new
    Panic.throw x
```
Easily reproducible via cli
```
built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/bin/enso --no-ir-caches --run ~/work/repos/enso/test-cases/thrownpe.enso 
Execution finished with an error: null
        at <enso> Panic.throw(Internal)
        at <enso> thrownpe.main(thrownpe.enso:6:5-17)
```

It's because NPEs message is null. (Hubert Plociniczak - Dec 20, 2022)
<hr />

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

IDE currently completely ignores `executionContext/executionStatus` messages without giving user any clue - even if the reported status signals an error.

### Checklist

Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [ ] All code conforms to the
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: app/gui/src/model/project.rs [hunk 1]
```diff
@@ -164,12 +164,17 @@ pub type Synchronized = synchronized::Project;
...
 pub enum Notification {
     /// One of the backend connections has been lost.
     ConnectionLost(BackendConnection),
     /// Indicates that the project VCS status has changed.
     VcsStatusChanged(VcsStatus),
+    /// Any message to be displayed as a notification
+    Message {
+        /// text of the message
+        message: String,
+    },
```
Comment (farmaazon) [inline app/gui/src/model/project.rs:177]: It's not needed, we already have such notifications in https://github.com/enso-org/enso/blob/20149c19bb0835cb0456dba3a548c125840c409a/app/gui/src/controller/ide.rs#L35

**Review by farmaazon — CHANGES_REQUESTED** [review]
I think sometimes this solution can rather confuse user than help him:
1. The errors reported in the execution status are still not displayed if warning happens to be first on the list
2. We do not inform the user about the place where error is produced. Theoretically it could be in the different file

Overall, I'd prefer to implement #5201 first, then make it properly: display a message with details section, and in details print all the information we get in executionStatus message.

The error log could remain. But then, let's print errors for all messages.

**Comment by jdunkerley** [comment]
Blocked until #5728 is completed, but should be done before design partners release.

**Comment by JaroslavTulach** [comment]
Original problem addressed by #5794. Closing.

**Pull request closed without merging**